lightautoml/ml_algo/torch_based/nn_models.py

```diff
@@ -975,3 +992,112 @@ def forward(self, embedded: torch.Tensor) -> torch.Tensor:
             mix = torch.sigmoid(self.mix)
             out = mix * out + (1 - mix) * self.mlp(embedded_2d)
         return out
+
+
+class FTTransformer(nn.Module):
+    """FT Transformer (https://arxiv.org/abs/2106.11959v2) from https://github.com/lucidrains/tab-transformer-pytorch/tree/main.
+
+    Args:
+            pooling: Pooling used for the last step.
+            n_out: Output dimension, 1 for binary prediction.
+            embedding_size: Embeddings size.
+            depth: Number of Attention Blocks inside Transformer.
+            heads: Number of heads in Attention.
+            attn_dropout: Post-Attention dropout.
+            ff_dropout: Feed-Forward Dropout.
+            dim_head: Attention head dimension.
+            return_attn: Return attention scores or not.
+            num_enc_layers: Number of Transformer layers.
+            device: Device to compute on.
+    """
+
+    def __init__(
+        self,
+        *,
+        pooling: str = "concat",
+        n_out: int = 1,
+        embedding_size: int = 16,
+        depth: int = 2,
+        heads: int = 1,
+        attn_dropout: float = 0.1,
+        ff_dropout: float = 0.1,
+        dim_head: int = 64,
+        return_attn: bool = False,
+        num_enc_layers: int = 2,
+        device: Union[str, torch.device] = "cuda:0",
+        **kwargs,
+    ):
+        super(FTTransformer, self).__init__()
+        self.return_attn = return_attn
+        # self.num_enc_layers = num_enc_layers
+        self.device = device
+        if pooling == "cls":
```

Comment:
Rewrite with factory pattern, i.e. you can use dict.

Reply:
done